src/utils.js

```diff
@@ -44,71 +44,125 @@ function dbExists (dbName) {
   })
 }
 
+/**
+ *
+ * @param {string} key - The key (/prefix/key)
+ * @param {string} nonce - The nonce is used to ensure privacy and avoid de-anonymization (hex string)
+ */
+const hashKey = async (key, nonce) => {
+  if (key === '/' || key === '') return '/'
+  const prefixes = key.split('/').filter(prefix => prefix !== '')
+  const hashedPrefixes = await Promise.all(prefixes.map(async (prefix) => hash256(`${nonce}${prefix}`)))
+  return hashedPrefixes.join('/')
+}
+
 /**
    * Get a value
    * @param {Object} db - The hyperDB instance
    * @param {CryptoKey} enckey - The enc/dec AES key
    * @param {string} key - The key name
+   * @param {string} nonce - The nonce as hex string
    * @returns {Promise}
    */
-const get = async (db, encKey, key) => {
+const get = async (db, encKey, nonce, key) => {
   if (!db) throw new MasqError(ERRORS.NO_DB)
   if (!encKey) throw new MasqError(ERRORS.NO_ENCRYPTION_KEY)
-  const node = await db.getAsync(key)
+  if (!nonce) throw new MasqError(ERRORS.NO_NONCE)
+  const hashedKey = await hashKey(key, nonce)
+
+  const node = await db.getAsync(hashedKey)
   if (!node) return null
   const dec = await decrypt(encKey, node.value)
-  return dec
+  // dec contains an object with key and value
+  return dec.value
 }
 
 /**
    * Put a new value in the db
    * @param {Object} db - The hyperDB instance
    * @param {CryptoKey} enckey - The enc/dec AES key
+   * @param {string} nonce - The nonce as hex string
    * @param {string} key - The key name
    * @param {any} value - The value to insert
    * @returns {Promise}
    */
-const put = async (db, encKey, key, value) => {
+const put = async (db, encKey, nonce, key, value) => {
   if (!db) throw new MasqError(ERRORS.NO_DB)
   if (!encKey) throw new MasqError(ERRORS.NO_ENCRYPTION_KEY)
-  const enc = await encrypt(encKey, value)
-  return db.putAsync(key, enc)
+  if (!nonce) throw new MasqError(ERRORS.NO_NONCE)
+
+  let sanitizedKey = key
+
+  if (sanitizedKey[0] === '/') {
+    sanitizedKey = sanitizedKey.slice(1)
+  }
+
+  if (sanitizedKey[sanitizedKey.length - 1] === '/') {
+    sanitizedKey = sanitizedKey.slice(0, -1)
+  }
```

Comment:
Add a comment to explain that the key stored in the value is sanitized to remove the '/' at the beginning and end of the '/' separated key.

Reply:
IIUC this sanitization is not needed, as `hashKey` will remove unnecessary slashes in all cases ?